docs/discussion/selection-api.md

```diff
@@ -0,0 +1,378 @@
+
+##### Option 3:
+```
+def SelectSources(match_func: Callable) -> None
+```
+Sends a request to aladin-lite to select all catalog sources that match
+the provided function 
+
+  **Parameter**   |**Type**   |**Description**
+  --------------- |---------- |-----------------------------------------------------
+  match_func      |lambda     |Lambda expression that defines that matching criteria for sources
+  --------------------------------------------------------------------------------
+
+#### SelectRegion
+```
+def SelectRegion(region: Region) -> None
+```
+This method triggers a selection by region event based on the coordinates 
+
+  **Parameter**   |**Type**                                                                                            |**Description**
+  --------------- |---------------------------------------------------------------------------------------------------| -------------------------------------------------------
+  region          |[Region](https://astropy-regions.readthedocs.io/en/stable/api/regions.Region.html#regions.Region)   |The region corresponding to the sky coordinates that the user would like to select sources within
+  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+#### SelectFootprint
+```
+def SelectFootprint(footprint: Footprint) -> None
+```
+This method triggers a selection event on a user provided footprint
+
+  **Parameter**   |**Type**    |**Description**
+  --------------- |----------- |---------------------------------------------
+  footprint       |Footprint   |The footprint the user would like to select
+
+  -------------------------------------------------------------------------
+
+# 3. aladin-lite interfaces
+
+## 3.1. Listeners
+
+Supported listeners that enable us to attach callback functions to
+events that occur within aladin-lite.
+
+  **Event Name**     |**Description**              |**Data**            |**Supported**
+  ------------------ |---------------------------- |------------------- |---------------
+  objectsSelected    |Fired when objects are selected by region, represents the individual objects selected |List of sources     |Existing
+  regionsSelected    |Fired when objects are selected by region, represents the region used to select the objects    |Shape of the region selected|New
+  objectClicked      |Fires when an object is clicked | Catalog object and xy coordinates of the click event | Existing
+  footprintClicked   |Fires when a footprint is clicked | Footprint object xy coordinates of the click event | Existing
+  -----------------------------------------------------------------------------------
+
+## 3.2. GET
+
+#### GetSelectedSources
+```
+/**
+ * @returns List[Sources]
+ */
+function GetSelectedSources()
+```
+Returns the list of currently selected source objects.
+
+#### GetSelectedRegion
```

Comment:
Related to the previous discussion about how regions are handled/maintained, maybe one possibility is that Aladin Lite doesn't maintain any persistence of regions, but just provides the building blocks for creating regions, leaving the management of persistent regions and applying them to selections to the Python layer.

Reply:
That's pretty much how the [demo implementation](https://github.com/pcuste1/aladin-lite/pull/1/files) I've provided functions! I added a "regionSelected" callback to Aaldin-lite that returns the building blocks for the region that was selected, and the [ipyaladin demo](https://github.com/pcuste1/ipyaladin/pull/3/files) uses those to generate astropy regions